README.md

```diff
@@ -1,655 +1,132 @@
+[Introduction](#introduction) | 
+[Documentation](#documentation) | 
+[Requirements](#requirements) | 
+[Usage](#usage) | 
+[License](/LICENSE)
 
-[https://www.youtube.com/watch?v=yGxeoAHlQOg](https://www.youtube.com/watch?v=yGxeoAHlQOg)
+<!-- TODO: Introduce CONTRIBUTING.md -->
 
-Become a Shopify App Developer
---------------------------------
-To become a Shopify App Developer, you'll need a [Shopify Partner account.](http://shopify.com/partners) If you don't have a Shopify Partner account, head to http://shopify.com/partners to create one before you start.
+## Introduction
 
-Once you have a Partner account, [create a new application in the Partner Dashboard](https://help.shopify.com/en/api/tools/partner-dashboard/your-apps) to get an API key and other API credentials.
+This gem includes a Rails engine, generators, modules, and mixins that help create Rails applications that work with Shopify APIs. The [Shopify App Rails engine](/docs/shopify_app/engine.md) provides all the code required to implement OAuth with Shopify. The [default Shopify App generator](/docs/shopify_app/generators.md#-environment-rails-generate-shopify_app) builds an app that can be embedded in the Shopify Admin and secures it with [session tokens](https://shopify.dev/concepts/apps/building-embedded-apps-using-session-tokens).
 
-To create an application for development set your new app's `App URL` to the URL provided by [your tunnel](#app-tunneling), ensuring that you use `https://`. If you are not planning to embed your app inside the Shopify admin or receive webhooks, set your redirect URL to `http://localhost:3000/` and the `Whitelisted redirection URL(s)` to contain `<App URL>/auth/shopify/callback`.
+## Documentation
 
-Installation
-------------
-To get started, add `shopify_app` to your Gemfile and run `bundle install`:
-
-``` sh
-# Create a new rails app
-$ rails new my_shopify_app
-$ cd my_shopify_app
-
-# Add the gem shopify_app to your Gemfile
-$ bundle add shopify_app
-```
+You can find documentation on gem usage, concepts, mixins, installation, and more in [`/docs`](/docs).
 
-Now we are ready to run any of the [generators](#generators) included with `shopify_app`. The following section explains the generators and what you can do with them.
+* Start with the [*Generators*](/docs/shopify_app/generators.md) document to learn more about the generators this gem offers.
+* Check out the [*Changelog*](/CHANGELOG.md) for notes on the latest gem releases.
+* See [*Troubleshooting*](/docs/Troubleshooting.md) for tips on common issues.
+* If you are looking to upgrade your Shopify App version to a new major release, see [*Upgrading*](/docs/Upgrading.md) for important notes on breaking changes.
 
+### Overview
 
-#### Rails Compatibility
+[Quickstart](/docs/Quickstart.md)
 
-The latest version of shopify_app is compatible with Rails `>= 5`. Use version `<= v7.2.8` if you need to work with Rails 4.
+[Releasing](/docs/Releasing.md)
```

Comment:
Does it make sense to include releasing instructions here? Not sure how relevant it is for non-Shopify developers.